examples/interactive/cli-platform/Env.roc

```diff
@@ -13,3 +13,90 @@ cwd =
+## (`�`).
+var : Str -> Task Str [VarNotFound]* [Env]*
+var = \name ->
+    Effect.envVar name
+    |> Effect.map (\result -> Result.mapErr result \{} -> VarNotFound)
+    |> InternalTask.fromEffect
+
+# ## Reads the given environment variable and attempts to decode it.
+# ##
+# ## The type being decoded into will be determined by type inference. For example,
+# ## if this ends up being used like a `Task U16 …` then the environment variable
+# ## will be decoded as a string representation of a `U16`.
+# ##
+# ##     getU16Var : Str -> Task U16 [VarNotFound, DecodeErr DecodeError]* [Env]*
+# ##     getU16Var = \var -> Env.decode var
+# ##     # If the environment contains a variable NUM_THINGS=123, then calling
+# ##     # (getU16Var "NUM_THINGS") would return a task which succeeds with the U16 number 123.
+# ##     #
+# ##     # However, if the NUM_THINGS environment variable was set to 1234567, then
+# ##     # (getU16Var "NUM_THINGS") would fail because that number is too big to fit in a U16.
+# ##
+# ## Supported types:
+# ## - strings
+# ## - numbers, as long as they contain only numeric digits, up to one `.`, and an optional `-` at the front for negative numbers
+# ## - comma-separated lists (of either strings or numbers), as long as there are no spaces after the commas
+# ##
+# ## Trying to decode into any other types will always fail with a `DecodeErr`.
+# decode : Str -> Task val [VarNotFound, DecodeErr DecodeError]* [Env]*
+#     | val has Decode
+# decode = \var ->
+#     Effect.envVar var
+#     |> InternalTask.fromEffect
+## Reads all the process's environment variables into a [Dict].
+##
+## If any key or value contains invalid Unicode, the [Unicode replacement character](https://unicode.org/glossary/#replacement_character)
+## (`�`) will be used in place of any parts of keys or values that are invalid Unicode.
+dict : Task (Dict Str Str) * [Env]*
+dict =
+    Effect.envDict
+    |> Effect.map Ok
+    |> InternalTask.fromEffect
+
+# ## Walks over the process's environment variables as key-value arguments to the walking function.
+# ##
+# ##     Env.walk "Vars:\n" \state, key, value ->
+# ##         "- \(key): \(value)\n"
+# ##     # This might produce a string such as:
+# ##     #
+# ##     #     """
+# ##     #     Vars:
+# ##     #     - FIRST_VAR: first value
+# ##     #     - SECOND_VAR: second value
+# ##     #     - THIRD_VAR: third value
+# ##     #
+# ##     #     """
+# ##
+# ## If any key or value contains invalid Unicode, the [Unicode replacement character](https://unicode.org/glossary/#replacement_character)
+# ## (`�`) will be used in place of any parts of keys or values that are invalid Unicode.
+# walk : state, (state, Str, Str -> state) -> Task state [NonUnicodeEnv state]* [Env]*
+# walk = \state, walker ->
+#     Effect.envWalk state walker
+#     |> InternalTask.fromEffect
+# TODO could potentially offer something like walkNonUnicode which takes (state, Result Str Str, Result Str Str) so it
```

Comment:
Do you mean
```suggestion
# TODO could potentially offer something like walkNonUnicode which takes (state, Result Str (List U8), Result Str (List U8)) so it
```

? I am assuming in the error case you present the list of bytes, maybe that's wrong tho